File: dayTen/Java/QuestionTwo.java
public class QuestionTwo {

    public static int [] sortArray(int[] arrayOfNumbers){
        int store = 0;
        
        for(int index = 0; index < arrayOfNumbers.length; index++){

            for(int count = 0; count < arrayOfNumbers.length; count++){
                if(arrayOfNumbers[index] < arrayOfNumbers[count]){
                    store = arrayOfNumbers[index];
                    arrayOfNumbers[index] = arrayOfNumbers[count];
                    arrayOfNumbers[count] = store;
                }
            }
        }
        return arrayOfNumbers;
    }

    public static int countSimilarIn(int[] firstArray, int [] secondArray){
    
        int [] sortedFirstArray = sortArray(firstArray);
        int [] sortedSecondArray = sortArray(secondArray);
        int counter = 0;
        int store = 0;
        
        for(int index = 0; index < sortedFirstArray.length; index++){
            
            for(int count = 0; count < sortedSecondArray.length; count++){
                if((sortedFirstArray[index] != store) && (sortedFirstArray[index] == sortedSecondArray[count]) ){
                    counter++;
                store = sortedFirstArray[index];
                }
            
            }
        }
        return counter;
    }
    
    public static int [] getSimilarIn(int [] firstArray, int [] secondArray){
        int [] newArray = new int[countSimilarIn(firstArray,secondArray)];
        
        int [] sortedFirstArray = sortArray(firstArray);
        int [] sortedSecondArray = sortArray(secondArray);
        int counter = 0;
        int store = 0;
        
        for(int index = 0; index < sortedFirstArray.length; index++){
            
            for(int count = 0; count < sortedSecondArray.length; count++){
                if((sortedFirstArray[index] != store) && (sortedFirstArray[index] == sortedSecondArray[count]) ){
                    newArray[counter] = sortedFirstArray[index];
                    counter++;
                    store = sortedFirstArray[index];
                }
            }
        }
        
        return newArray;
    }
}

File: dayTen/Javascript/QuestionTwo.js
let sortArray=(arrayOfNumbers)=>{
    let store = 0;
    
    for(let index = 0; index < arrayOfNumbers.length; index++){

        for(let count = 0; count < arrayOfNumbers.length; count++){
            if(arrayOfNumbers[index] < arrayOfNumbers[count]){
                store = arrayOfNumbers[index];
                arrayOfNumbers[index] = arrayOfNumbers[count];
                arrayOfNumbers[count] = store;
            }
        }
    }
    return arrayOfNumbers;
}

let countSimilarIn=(firstArray, secondArray)=>{

    let sortedFirstArray = sortArray(firstArray);
    let sortedSecondArray = sortArray(secondArray);
    let counter = 0;
    let store = 0;
    
    for(let index = 0; index < sortedFirstArray.length; index++){
        
        for(let count = 0; count < sortedSecondArray.length; count++){
            if((sortedFirstArray[index] != store) && (sortedFirstArray[index] == sortedSecondArray[count]) ){
                counter++;
                store = sortedFirstArray[index];
            }
        }
    }
    return counter;
}

let getSimilarIn=(firstArray, secondArray)=>{
    let newArray = new Array([countSimilarIn(firstArray,secondArray)]).fill(0);
    
    let sortedFirstArray = sortArray(firstArray);
    let sortedSecondArray = sortArray(secondArray);
    let counter = 0;
    let store = 0;
    
    for(let index = 0; index < sortedFirstArray.length; index++){
        
        for(let count = 0; count < sortedSecondArray.length; count++){
            if((sortedFirstArray[index] != store) && (sortedFirstArray[index] == sortedSecondArray[count]) ){
                newArray[counter] = sortedFirstArray[index];
                counter++;
                store = sortedFirstArray[index];
            }
        }
    }
    
    return newArray;
}



firstArray = [1,2,2,1];
secondArray = [2,2];

console.log(getSimilarIn(firstArray,secondArray));
